contribute/architecture/backend/services.md

```diff
@@ -2,71 +2,136 @@
 
 A Grafana _service_ encapsulates and exposes application logic to the rest of the application, through a set of related operations.
 
-Before a service can start communicating with the rest of Grafana, it needs to be registered in the _service registry_.
-
-The service registry keeps track of all available services during runtime. On start-up, Grafana uses the registry to build a dependency graph of services, a _service graph_.
+Grafana uses [Wire](https://github.com/google/wire), which is a code generation tool that automates connecting components using [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection). Dependencies between components are represented in Wire as function parameters, encouraging explicit initialization instead of global variables.
 
 Even though the services in Grafana do different things, they share a number of patterns. To better understand how a service works, let's build one from scratch!
 
-## Create a service
-
-To start building a service:
+Before a service can start communicating with the rest of Grafana, it needs to be registered with Wire, see `ProvideService` factory function/method in the service example below and how it's being referenced in the wire.go example below.
 
-- Create a new Go package `mysvc` in the [pkg/services](/pkg/services) directory.
-- Create a `service.go` file inside your new directory.
+When Wire is run it will inspect the parameters of `ProvideService` and make sure that all it's dependencies has been wired up and initialized properly.
 
-All services need to implement the [Service](https://godoc.org/github.com/grafana/grafana/pkg/registry#Service) interface:
+**Service example:**
 
 ```go
-type MyService struct {
-}
+package example
 
-func (s *MyService) Init() error {
-    return nil
+// Service service is the service responsible for X, Y and Z.
+type Service struct {
+    logger   log.Logger
+    cfg      *setting.Cfg
+    sqlStore *sqlstore.SQLStore
 }
-```
 
-The `Init` method is used to initialize and configure the service to make it ready to use. Services that return an error halt Grafana's startup process and cause the error to be logged as it exits.
+// ProvideService provides Service as dependency for other services.
+func ProvideService(cfg *setting.Cfg, sqlStore *sqlstore.SQLStore) (*Service, error) {
```

Comment:
I think we should extend the description to say that ProvideService can accept dynamic number of arguments where arguments are other existing service implementations. Just not very obvious from the first look.

Also, related thing - previously there was a possibility to set a priority for a service - it's now gone and determined by an order inside wireSets?

Reply:
Will fix this later in a separate PR.

Regarding priority, there's no need for that with Wire. Wire will make sure to resolve the service dependency graph for us so that services are initialized in the correct/expected. order